### ISSUE TITLE
Standalone APK built with Expo SDK 47 crashes on specific device

### ISSUE THREAD

**Issue by otekdo** [expo/expo#20024]
### Summary

Hello, I am getting issues when i try to use the production mode `(APK)` of my first expo app.

- The development environment works nice with **Android Emulator** + Expo Go (Never got error logs from console)
- The development environment works nice with **iOS device** + Expo Go (Never got error logs from console)

The production build works nicely on my android emulator.

But the production build **does NOT works** on my physical device Galaxy Note 8 (Android 7.1.1)

`Everything was going ok, until I updated my app to SDK 47`

### What platform(s) does this occur on?

Android

### Environment

    System:
      OS: Windows 10 10.0.19045
    Binaries:
      Node: 16.15.1 
      Yarn: 1.22.19 
      npm: 8.11.0 
    IDEs:
      Android Studio: AI-213.7172.25.2113.9123335     
    npmPackages:
      expo: ~47.0.3 => 47.0.3 
      react: 18.1.0 => 18.1.0 
      react-dom: 18.1.0 => 18.1.0 
      react-native: 0.70.5 => 0.70.5 
      react-native-web: ~0.18.7 => 0.18.10 
      Expo Workflow: managed

    Physical Device:
      Samsung Galaxy Note 8 (SAMSUNG-SM-N950U) (Android 7.1.1)

### Minimal reproducible example

I just need to open the production app on my physical device, it will crash.
Everything works fine in the emulator.

This is the `package.json`:
```json
{
  "name": "my-nice-app",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "local-build": "eas build -p android --profile preview"
  },
  "dependencies": {
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@react-navigation/material-bottom-tabs": "^6.2.4",
    "@react-navigation/native": "^6.0.13",
    "@react-navigation/native-stack": "^6.9.1",
    "expo": "~47.0.3",
    "expo-status-bar": "~1.4.2",
    "react": "18.1.0",
    "react-dom": "18.1.0",
    "react-native": "0.70.5",
    "react-native-paper": "5.0.0-rc.9",
    "react-native-safe-area-context": "4.4.1",
    "react-native-screens": "~3.18.0",
    "react-native-web": "~0.18.7",
    "zustand": "^4.1.4"
  },
  "devDependencies": {
    "@babel/core": "^7.19.3",
    "@types/react": "~18.0.24",
    "@types/react-native": "~0.70.6",
    "typescript": "^4.6.3"
  },
  "private": true
}
```
Source code is public: https://github.com/otekdo/my-nice-app/blob/master/package.json

### Additional actions
I already tried npx expo start --no-dev --minify expo go, works nicely on the emulator

**Comment by RohovDmytro** [comment]
I've also experience crashed on older version of Android. The expo team just released an update. Bumping the expo to 47.0.5 fixed an issue for me.

**Comment by otekdo** [comment]
The issue is fixed after updating expo to ``@~47.0.5``, thanks @brentvatne
